### PR TITLE
Plans: Remove references to Happy Chat

### PR DESCRIPTION
**Pull request by klimeryk**
Part of efforts to sunset Happy Chat and remove all references: pdm0RZ-BU-p2

## Proposed Changes

* Removes FAQ component from Plans page - [it's confirmed dead](https://github.com/Automattic/wp-calypso/pull/78262#issuecomment-1593144469).

## Testing Instructions

- Do a code search and confirm that this is indeed dead code.
- Although E2E tests should take care of this, spot check a few flows: /start, /plans/<site slug>, /setup/newsletter/plans.

### CODE REVIEW

**Comment by klimeryk** [comment]
Looks like the `showFAQ` flag was indeed hardcoded in https://github.com/Automattic/wp-calypso/pull/77493 from a previous value of `showFAQ: !! props.showFAQ && ! is2023PricingGridVisible,`. @aneeshd16, @chriskmnds, should the `WpcomFAQ` component then be simply removed? If it's not gonna be used, it's dead code, but still requires maintenance, as seen by this PR.

**Comment by aneeshd16** [comment]
Thanks for the ping @klimeryk! I checked the usage of the component, and I think we can remove it safely. I'll put up a PR for this shortly.

File: client/my-sites/plans-features-main/index.jsx [hunk 1]
```diff
@@ -26,7 +26,6 @@ import {
 	getPlanPath,
 	GROUP_WPCOM,
 	PLAN_PERSONAL,
-	TITAN_MAIL_MONTHLY_SLUG,
```
Comment (AtrumGeost) [inline client/my-sites/plans-features-main/index.jsx:29]: It looks like this was removed intentionally, I'm just curious why since doesn't seem related to Happy Chat orthe wpcom-faq implementations.

**Review by AtrumGeost — COMMENTED** [review]
Thanks for working on this 🙏 It removes noise and makes everyone's work easier.

I searched through the modified files and the references are properly removed. I just left a question about something that dragged my curiosity.

File: client/my-sites/plans-features-main/index.jsx (hunk 1 above)
Comment (klimeryk) [reply client/my-sites/plans-features-main/index.jsx:29]: It goes back to this line:
https://github.com/Automattic/wp-calypso/pull/78262/files#diff-2545c62593dd002397248377b1c0feb643bd2acc7694054ce094997b1c029aceL422

Which was removed because that whole section is dead. `titanMonthlyRenewalCost` is set on this line:
https://github.com/Automattic/wp-calypso/pull/78262/files#diff-2545c62593dd002397248377b1c0feb643bd2acc7694054ce094997b1c029aceL557

Which depends on the `TITAN_MAIL_MONTHLY_SLUG` constant. Since it was not longer referenced, I removed it. Hope that clears it up!
